modules/core/misc/java/src/java/CustomCleaner.java

```diff
@@ -0,0 +1,149 @@
+package org.opencv.core;
+
+import java.lang.ref.PhantomReference;
+import java.lang.ref.ReferenceQueue;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * This class is a Java 8+ implementation Cleaner similar to java.lang.ref.Cleaner available
+ * from Java 9.
+ * <p>
+ * This implementation replace finalize() method that is deprecated since Java 9 and for removal
+ * since Java 18
+ * <p>
+ * When OpenCV has Java 8 as its minimum version, this class can be removed and replaced by java.lang.ref.Cleaner.
+ * In Mat, <code>public static final Cleaner cleaner = Cleaner.create();</code>
+ */
+public final class CustomCleaner {
+
+    final PhantomCleanable phantomCleanableList;
+
+    // The ReferenceQueue of pending cleaning actions
+    final ReferenceQueue<Object> queue;
+
+    private CustomCleaner() {
+        queue = new ReferenceQueue<>();
+        phantomCleanableList = new PhantomCleanable();
+    }
+
+    public static CustomCleaner create() {
+        CustomCleaner customCleaner = new CustomCleaner();
+        customCleaner.start();
+        return customCleaner;
+    }
+
+    public Cleanable register(Object obj, Runnable action) {
+        return new PhantomCleanable(Objects.requireNonNull(obj), Objects.requireNonNull(action));
+    }
+
+    private void start() {
+        new PhantomCleanable(this, null);
+        Thread thread = new CleanerThread(() -> {
```

Comment:
`new Thread` captures `Thread.inheritedAccessControlContext` so the code will likely run into memory leak like in https://stackoverflow.com/questions/78163385/does-postgres-jdbc-lazycleaner-cause-a-classloader-memory-leak/79835942